core/lib/dal/src/connection/mod.rs

```diff
@@ -68,85 +60,116 @@ impl ConnectionPoolBuilder {
             statement_timeout = self.statement_timeout
         );
         Ok(ConnectionPool {
+            database_url: self.database_url.clone(),
             inner: pool,
             max_size: self.max_size,
         })
     }
 }
 
-/// Constructs a new temporary database (with a randomized name)
-/// by cloning the database template pointed by TEST_DATABASE_URL env var.
-/// The template is expected to have all migrations from dal/migrations applied.
-/// For efficiency, the Postgres container of TEST_DATABASE_URL should be
-/// configured with option "fsync=off" - it disables waiting for disk synchronization
-/// whenever you write to the DBs, therefore making it as fast as an in-memory Postgres instance.
-/// The database is not cleaned up automatically, but rather the whole Postgres
-/// container is recreated whenever you call "zk test rust".
-pub(super) async fn create_test_db() -> anyhow::Result<url::Url> {
-    use rand::Rng as _;
-    use sqlx::{Connection as _, Executor as _};
-
-    const PREFIX: &str = "test-";
-
-    let db_url = get_test_database_url().unwrap();
-    let mut db_url = url::Url::parse(&db_url)
-        .with_context(|| format!("{} is not a valid database address", db_url))?;
-    let db_name = db_url
-        .path()
-        .strip_prefix('/')
-        .with_context(|| format!("{} is not a valid database address", db_url.as_ref()))?
-        .to_string();
-    let db_copy_name = format!("{PREFIX}{}", rand::thread_rng().gen::<u64>());
-    db_url.set_path("");
-    let mut attempts = 10;
-    let mut conn = loop {
-        match sqlx::PgConnection::connect(db_url.as_ref()).await {
-            Ok(conn) => break conn,
-            Err(err) => {
-                attempts -= 1;
-                if attempts == 0 {
-                    return Err(err).context("sqlx::PgConnection::connect()");
-                }
-            }
-        }
-        tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-    };
-    conn.execute(
-        format!("CREATE DATABASE \"{db_copy_name}\" WITH TEMPLATE \"{db_name}\"").as_str(),
-    )
-    .await
-    .context("failed to create a temporary database")?;
-    db_url.set_path(&db_copy_name);
-    Ok(db_url)
-}
-
 #[derive(Clone)]
 pub struct ConnectionPool {
     pub(crate) inner: PgPool,
+    database_url: String,
     max_size: u32,
 }
 
 impl fmt::Debug for ConnectionPool {
     fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
         formatter
             .debug_struct("ConnectionPool")
+            .field("database_url", &self.database_url)
```

Comment:
It was previously discussed that `database_url` can contain sensitive information (e.g. database password), so it shouldn't be printed lightheartedly. Probably it makes sense to write a comment stating that this field is omitted intentionally.